material_studies/rally/lib/finance.dart

```diff
@@ -0,0 +1,332 @@
+    @required this.subtitle,
+    @required this.amount,
+    @required this.suffix,
+  });
+
+  final Color indicatorColor;
+  final double indicatorFraction;
+  final String title;
+  final String subtitle;
+  final double amount;
+  final Widget suffix;
+
+  @override
+  Widget build(BuildContext context) {
+    return FlatButton(
+      onPressed: () {
+        Navigator.push(
+          context,
+          MaterialPageRoute<FinancialEntityCategoryDetailsPage>(
+            builder: (context) => FinancialEntityCategoryDetailsPage(),
+          ),
+        );
+      },
+      child: SizedBox(
+        height: 68,
+        child: Column(
+          children: [
+            Expanded(
+              child: Row(
+                children: [
+                  Padding(
+                    padding: EdgeInsets.only(left: 12, right: 12),
+                    child: VerticalFractionBar(
+                      color: indicatorColor,
+                      fraction: indicatorFraction,
+                    ),
+                  ),
+                  Column(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    crossAxisAlignment: CrossAxisAlignment.start,
+                    children: [
+                      Text(
+                        title,
+                        style: Theme.of(context)
+                            .textTheme
+                            .body1
+                            .copyWith(fontSize: 16.0),
+                      ),
+                      Text(
+                        subtitle,
+                        style: Theme.of(context)
+                            .textTheme
+                            .body1
+                            .copyWith(color: RallyColors.gray60),
+                      ),
+                    ],
+                  ),
+                  Spacer(),
+                  Text(
+                    '\$ ' + Formatters.usd.format(amount),
+                    style: Theme.of(context)
+                        .textTheme
+                        .body2
+                        .copyWith(fontSize: 20.0, color: RallyColors.gray),
+                  ),
+                  SizedBox(width: 32.0, child: suffix),
+                ],
+              ),
+            ),
+            Divider(
+              height: 1,
+              indent: 16,
+              endIndent: 16,
+              color: Color(0xAA282828),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+/// Data model for [FinancialEntityCategoryView].
+class FinancialEntityCategoryModel {
+  final Color indicatorColor;
+  final double indicatorFraction;
+  final String title;
+  final String subtitle;
+  final double usdAmount;
+  final Widget suffix;
+
+  const FinancialEntityCategoryModel(
+    this.indicatorColor,
+    this.indicatorFraction,
+    this.title,
+    this.subtitle,
+    this.usdAmount,
+    this.suffix,
+  );
+}
+
+FinancialEntityCategoryView fromAccountItem(AccountData model, int i) {
```

Comment:
These three methods need to be renamed similar to the ones below (`build...`).

Reply:
Done